### PR TITLE
Update cryptoauthlib to version 3.3.0, convert to git submodule

### PR DESCRIPTION
**Pull request by paulbartell**
Update cryptoauthlib to the v3.3.0 release from microchip.

This PR removes the copy of cryptoauthlib in this repository and replaces it with a git submodule to microchip's official [cryptoauthlib github repository](https://github.com/MicrochipTech/cryptoauthlib.git).

Based on a submission from microchip in #3129

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: vendors/microchip/boards/ecc608a_plus_winsim/aws_demos/config_files/atca_config.h [hunk 1]
```diff
@@ -28,17 +28,30 @@
...
 
-/** Use I2C */
+ /** Use HID */
 #define ATCA_HAL_KIT_HID
+#define ATCA_ATECC608A_SUPPORT
 
 /** Use the following address for ECC devices */
-#define ATCA_I2C_ECC_ADDRESS    0x6C
+#define ATCA_I2C_ECC_ADDRESS           0xC0
+#define ATCA_DEVICE_PRIVATE_KEY_SLOT   0x00
```
Comment (ravibhagavandas) [inline vendors/microchip/boards/ecc608a_plus_winsim/aws_demos/config_files/atca_config.h:37]: Does it need update for EPS32 ECC608 config files ?
https://github.com/aws/amazon-freertos/blob/main/vendors/espressif/boards/esp32/aws_demos/config_files/ecc608a_pkcs11_config/atca_config.h

Comment (paulbartell) [reply vendors/microchip/boards/ecc608a_plus_winsim/aws_demos/config_files/atca_config.h:37]: I believe the address should stay the same for other platforms. Perhaps we should change this board name to ecc608b_plus_winsim.

Comment (ravibhagavandas) [reply vendors/microchip/boards/ecc608a_plus_winsim/aws_demos/config_files/atca_config.h:37]: The macro like `ATCA_DEVICE_PRIVATE_KEY_SLOT ` needs to be defined in `https://github.com/aws/amazon-freertos/blob/main/vendors/espressif/boards/esp32/aws_demos/config_files/ecc608a_pkcs11_config/atca_config.h` or in a common header file. Its been used in `libraries/abstractions/pkcs11/ecc608a/atca_cert_chain.c` which I guess is common for both platforms.

File: vendors/microchip/secure_elements/lib/hal/hal_esp32_i2c.c [hunk 2]
```diff
@@ -1,311 +0,0 @@
-/*
```
[the remark this answers is not in the document]
Comment (paulbartell) [reply vendors/microchip/secure_elements/lib/hal/hal_esp32_i2c.c:1]: It has moved to a different directly. Will update.

**Review by cobusve — DISMISSED** [review]

**Pull request closed without merging**